scrapy_poet/injection.py

```diff
@@ -72,25 +73,22 @@ def load_providers(self, default_providers: Optional[Mapping] = None):  # noqa:
             self.providers
         )
 
-    def close(self) -> None:  # noqa: D102
-        if self.cache:
-            self.cache.close()
-
     def init_cache(self):  # noqa: D102
-        self.cache = None
-        cache_filename = self.crawler.settings.get("SCRAPY_POET_CACHE")
-        if cache_filename and isinstance(cache_filename, bool):
-            cache_filename = os.path.join(
-                get_scrapy_data_path(createdir=True), "scrapy-poet-cache.sqlite3"
-            )
-        if cache_filename:
-            compressed = self.crawler.settings.getbool("SCRAPY_POET_CACHE_GZIP", True)
+        self.cache = {}
+        cache_path = self.crawler.settings.get("SCRAPY_POET_CACHE")
+
+        # SCRAPY_POET_CACHE: True
+        if cache_path and isinstance(cache_path, bool):
+            cache_path = get_scrapy_data_path(createdir=True)
```

Comment:
I wonder if this should have some more specific name, like the old path including "scrapy-poet-cache" in the file name.